File: src/main/scala/binaryTreeScala/UserFactory.scala
package binaryTreeScala

import java.util
import scala.jdk.CollectionConverters._

class UserFactory {
  def getTypeNameList: util.List[String] = List("Point", "Fraction").asJava

  def getBuilderByName(name: String): UserType = name match {
    case "Point"    => new Point
    case "Fraction" => new Fraction
    case _          => null
  }
}

File: src/main/java/binaryTree/UserFactory.java
package binaryTree;

import java.util.ArrayList;
import java.util.List;

public class UserFactory {
    public ArrayList<String> getTypeNameList() {
        return new ArrayList<>(List.of("Point", "Fraction"));
    }

    public UserType getBuilderByName(String name){
        return switch (name) {
            case "Point" -> new Point();
            case "Fraction" -> new Fraction();
            default -> null;
        };

    }
}
